test/integration/test_slct_manager.py

```diff
@@ -0,0 +1,193 @@
+)
+def test_upload(slct_manager: SlctManager, itde):
+    slct_manager.language_alias = "my_python"
+    slct_manager.upload()
+    assert slct_manager.activation_key == "my_python=localzmq+protobuf:///bfsdefault/default/container/template-Exasol-all-python-3.10-release-my_python?lang=python#buckets/bfsdefault/default/container/template-Exasol-all-python-3.10-release-my_python/exaudf/exaudfclient_py3"
+
+
+@pytest.mark.dependency(
+    name="append_custom_packages", depends=["upload_slc"]
+)
+def test_append_custom_packages(slct_manager: SlctManager, custom_packages: List[Tuple[str, str, str]]):
+    slct_manager.append_custom_packages([PipPackageDefinition(pkg, version) for pkg, version, _ in custom_packages])
+    with open(slct_manager.slc_dir.custom_pip_file, "r") as f:
+        pip_content = f.read()
+        for custom_package, version, _ in custom_packages:
+            assert f"{custom_package}|{version}" in pip_content
+
+
+@pytest.mark.dependency(
+    name="upload_slc_with_new_packages", depends=["append_custom_packages"]
+)
+def test_upload_slc_with_new_packages(slc_secrets: Secrets, slct_manager: SlctManager,
+                                      custom_packages: List[Tuple[str, str, str]]):
+    slct_manager.language_alias = "my_new_python"
+    slct_manager.upload()
+    assert slct_manager.activation_key == "my_new_python=localzmq+protobuf:///bfsdefault/default/container/template-Exasol-all-python-3.10-release-my_new_python?lang=python#buckets/bfsdefault/default/container/template-Exasol-all-python-3.10-release-my_new_python/exaudf/exaudfclient_py3"
+
+
+@pytest.mark.dependency(
+    name="udf_with_new_packages", depends=["upload_slc_with_new_packages"]
+)
+def test_udf_with_new_packages(slc_secrets: Secrets, slct_manager: SlctManager,
+                               custom_packages: List[Tuple[str, str, str]]):
+    import_statements = "\n".join(f"    import {module}" for pkg, version, module in custom_packages)
+    udf = textwrap.dedent(f"""
+CREATE OR REPLACE {slct_manager.language_alias} SET SCRIPT test_custom_packages(i integer)
+EMITS (o VARCHAR(2000000)) AS
+def run(ctx):
+{import_statements}
+
+    ctx.emit("success")
+/
+        """)
+    con = open_pyexasol_connection_with_lang_definitions(slc_secrets)
+    try:
+        con.execute("CREATE SCHEMA TEST")
+        con.execute(udf)
+        res = con.execute("select test_custom_packages(1)")
+        rows = res.fetchall()
+        assert rows == [('success',)]
+    finally:
+        con.close()
+
+
+@pytest.mark.dependency(
+    name="test_old_alias", depends=["udf_with_new_packages"]
+)
+def test_old_alias(slc_secrets: Secrets, slct_manager: SlctManager):
+
+    udf = textwrap.dedent(f"""
+CREATE OR REPLACE my_python SET SCRIPT test_old_slc(i integer)
+EMITS (o VARCHAR(2000000)) AS
+def run(ctx):
+    ctx.emit("success")
+/
+        """)
+    con = open_pyexasol_connection_with_lang_definitions(slc_secrets, schema='TEST')
```

Comment:
having a `connection` fixture would be a possible simiplification

Reply:
Would not work in this case, as I have these dependencies to the other tests.